orttraining/orttraining/test/training_ops/cpu/tensor/gather_nd_grad_op_test.cc

```diff
@@ -20,8 +19,20 @@ TEST(GatherNDGradOpTest, GatherNDGrad_slice_float_int64_t_batch_dims_1) {
   test.Run();
 }
 
+TEST(GatherNDGradOpTest, GatherNDGrad_slice_float_int64_t_batch_dims_zero_negative_indices) {
+  OpTester test("GatherNDGrad", 1, kMSDomain);
+  test.AddAttribute<int64_t>("batch_dims", 0);
+  test.AddInput<int64_t>("shape", {3}, {2LL, 2LL, 3LL});
+  test.AddInput<int64_t>("indices", {2, 2}, {-2LL, -1LL, -1LL, -2LL});
+  test.AddInput<float>("update", {2, 3}, ValueRange(6, 1.0f));
+  test.AddOutput<float>("output", {2, 2, 3}, {0, 0, 0, 1, 2, 3, 4, 5, 6, 0, 0, 0});
+  test.Run();
+}
+
 TEST(GatherNDGradOpTest, GatherNDGrad_slice_double_int32_t_batch_dims_3) {
-  if (!HasCudaEnvironment(600 /*min_cuda_architecture*/)) return;
+  if (NeedSkipIfCudaArchLowerThan(600)) {
```

Comment:
I remember "#if __CUDA_ARCH__ >= xxx" won't work properly in .cc. since you are on it, although it's out of scope, would it be ok that we also clean up the code in gather_grad_op_test.cc and gather_grad.cc using the same method? Or we can do this in separate PR. @weixingzhang

Reply:
ya, make sense! yes let me see whether I can fix them...